### PR TITLE
Sign up page

### PR DESCRIPTION
**Pull request by nina-dk**
Notes:

- I've updated the db schema so you'll need to reset the db before running the app.
- I added an email field in User which I think should also be made UNIQUE, but I haven't done that yet.
- Given the changes, more tests are needed for the `users` API. I'll create a separate issue for that.
- I changed the colour palette and some other things on both the sign-up and login pages quite a bit. I've left some comments in the CSS for you to view a couple of alternatives. **DO** let me know if you hate it, or want to switch back to the old design, or what not.
- I didn't implement that but I recommend we do away with the image in mobiles. It takes too much space, doesn't seem professional to me and increases loading time, too.
- The "Neighborhood" title I expect to be replaced by the app logo when that happens so haven't spent too much time styling it.
- This PR resolves issues #103 and #106.

### CODE REVIEW

File: apps/frontend/src/services/signUp.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+async function signUp(signUpData: SignUpData) {  
+  const newUser: UserWithoutPasswordHash = await axios.post(baseURL, signUpData);
+  if (newUser) {
+    const loginData: LoginData = {
+      username: signUpData.username,
+      password: signUpData.password,
+    }
+
+    // there is duplication from the login action
+    // is there a better way to do this?
```
Comment (radu-constantin) [inline apps/frontend/src/services/signUp.ts:18]: I don't understand. Can you explain please?

**Review by radu-constantin — APPROVED** [review]
Antonina, I hope one day I will write code as elegant as you do.
Great job as usual, I only have one question in the comment section.

File: apps/frontend/src/utils/auth.ts [hunk 2]
```diff
@@ -45,8 +45,22 @@ export function checkAuthLoader() {
...
+ * - else, returns `null`
+ * @returns
+ */
+export function redirectLoggedInUser() {
+  const user = getStoredUser();
+
+  if (user) return redirect('/');
+
+  return null;
+}
```
[the remark this answers is not in the document]
Comment (nina-dk) [reply apps/frontend/src/utils/auth.ts:66]: That's why I removed the code you're asking about above. I had added this (or someone else did, not sure) back then for users navigating to the login page while already being logged in on that device. Now this is not possible thanks to this loader.

File: apps/frontend/src/services/signUp.ts (hunk 1 above)
Comment (nina-dk) [reply apps/frontend/src/services/signUp.ts:18]: sure so what I'm doing there is after a successful sign-up, I log the user in using the `login` service and basically, repeating the actions that happen in the login action in `LoginPage.tsx`. I wonder if there is a way to achieve this without duplicating the code from the login action.

File: apps/frontend/src/services/login.ts [hunk 3]
```diff
@@ -1,18 +1,10 @@
...
 async function login(loginData: LoginData) {  
-  const headers: { authorization?: string } = {};
-  let { user }: { user?: string } = localStorage as StorageWithUser;
-
-  if (user) {
-    const userObj: UserInfo = JSON.parse(user);
-    headers.authorization = `Bearer ${userObj.token}`;
-  }
-
-  const response = await axios.post(baseURL, loginData, { headers });
```
[the remark this answers is not in the document]
Comment (nina-dk) [reply apps/frontend/src/services/login.ts:15]: Answered below.

File: apps/frontend/src/utils/auth.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (radu-constantin) [reply apps/frontend/src/utils/auth.ts:66]: Got it, thanks!

**Pull request merged**